From 009a243743bfb65448da0b0dae44361b3b0ef592 Mon Sep 17 00:00:00 2001
From: Richard Liu <r66033@freescale.com>
Date: Wed, 14 Oct 2015 09:55:58 +0800
Subject: [PATCH 5/6] MA-7102 fix netd build error on android 6.0 Mashmallow

fix build error caused by the conflict of fsl change and
android 6.0 Mashmallow.

Signed-off-by: Richard Liu <r66033@freescale.com>
---
 server/SoftapController.cpp | 13 +++++++++----
 1 file changed, 9 insertions(+), 4 deletions(-)

diff --git a/server/SoftapController.cpp b/server/SoftapController.cpp
index 03566dd..c24ae5e 100644
--- a/server/SoftapController.cpp
+++ b/server/SoftapController.cpp
@@ -158,10 +158,15 @@ int SoftapController::setSoftap(int argc, char *argv[]) {
             channel = AP_CHANNEL_DEFAULT;
     }
 
-    asprintf(&wbuf, "interface=%s\ndriver=nl80211\nctrl_interface="
-            "/data/misc/wifi/hostapd\nssid=%s\nchannel=%d\nieee80211n=1\n"
-            "hw_mode=g\nignore_broadcast_ssid=%d\n",
-            argv[2], argv[3], channel, hidden);
+    std::string wbuf(StringPrintf("interface=%s\n"
+            "driver=nl80211\n"
+            "ctrl_interface=/data/misc/wifi/hostapd\n"
+            "ssid=%s\n"
+            "channel=%d\n"
+            "ieee80211n=1\n"
+            "hw_mode=%c\n"
+            "ignore_broadcast_ssid=%d\n",
+            argv[2], argv[3], channel, (channel <= 14) ? 'g' : 'a', hidden));
 
     std::string fbuf;
     if (argc > 7) {
-- 
1.8.0

